AmplifyPlugins/Core/AWSPluginsCore/Model/Support/Model+GraphQL.swift

```diff
@@ -59,6 +59,8 @@ extension Model {
                             input.updateValue(nil, forKey: fieldName)
                         }
                     }
+                } else if case .collection = modelField.type { // skip all "has-many"
+                    continue
```

Comment:
bug where collections's are being set to `nil` in the GraphQL input, so for example, when creating a Post and saving it through DataStore. the sync process will translate the post to the GraphQL input variables containing
```
{
  "id": "postid123",
   "comments": null
}
```

The problem is this fails as there's no input type "comments". We should skip has-many relationships like in this code snippet. This can be done in `main` and `v1`

Reply:
https://github.com/aws-amplify/amplify-swift/issues/2483